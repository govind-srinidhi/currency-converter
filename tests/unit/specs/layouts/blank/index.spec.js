/**
 * @file It contains unit tests for blank layout.
 */
 import { mount, createLocalVue } from "@vue/test-utils"
 import Vuetify from "vuetify"
 import Blank from "@/layouts/blank"
 import Vuex from "vuex"

 let wrapper

 const currencySnackbarNotification = () => wrapper.find("currency-snackbar-notification-stub")

 let store

 const state = {
  notification: undefined
 }

 const mutations = {
  setNotification: (state, notification) => {
    state.notification = notification
  }
 }

 const getters = {
  "shared/notification": state => state.notification,
 }

 const actions = {
  "shared/suppressNotification": jest.fn()
 }

 const mountBlank = () => {
   const localVue = createLocalVue()

   localVue.use(Vuetify)
   localVue.use(Vuex)

   store = new Vuex.Store({
    state,
    getters,
    mutations,
    actions
   })
 
   const vuetify = new Vuetify()
 
   wrapper = mount(Blank, {
     sync: false,
     stubs: ["currency-snackbar-notification"],
     localVue,
     store,
     vuetify
   })
 }

 
 describe("blank layout", () => {
   beforeEach(() => {
     mountBlank()
   })
 
   afterEach(() => {
     jest.clearAllMocks()
   })
 
   it("renders the blank layout", () => {
     expect(wrapper).toBeTruthy()
   })
 
   it("renders elements correctly", async () => {
     expect(currencySnackbarNotification().exists()).toBeFalsy()
   })

   it("notification to be rendered, whenever notification is called", async () => {
     store.commit("setNotification", {
       type: "error",
       icon: "mdi-alert",
       text: "error notification"
     })
     await wrapper.vm.$nextTick()
     expect(currencySnackbarNotification().exists()).toBeTruthy()
     currencySnackbarNotification().vm.$emit("close")
     await wrapper.vm.$nextTick()
     expect(actions["shared/suppressNotification"]).toHaveBeenCalledTimes(1)
   })

   it("matches the default snapshot", async () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
 })