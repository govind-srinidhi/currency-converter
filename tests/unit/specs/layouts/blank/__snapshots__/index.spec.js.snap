// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blank layout matches the default snapshot 1`] = `
<div data-app="true" class="v-application v-application--is-ltr theme--light" id="app">
  <div class="v-application--wrap">
    <main class="v-main align-center" style="padding: 0px 0px 0px 0px;">
      <div class="v-main__wrap">
        <currency-snackbar-notification-stub pnotification="[object Object]"></currency-snackbar-notification-stub>
      </div>
    </main>
  </div>
</div>
`;
